QuizApp/QuizApp/Client/Login/LoginClient.swift

```diff
@@ -15,50 +15,19 @@ private struct LoginRequest: Codable {
 
 class LoginClient: LoginClientProtocol {
 
-    private let baseUrl: String
+    private let path = "/v1/login"
+    private let networkClient: NetworkClientProtocol
 
-    init(baseUrl: String) {
-        self.baseUrl = baseUrl
+    init(networkClient: NetworkClientProtocol) {
+        self.networkClient = networkClient
     }
 
     func login(username: String, password: String) async throws -> LoginResponseClientModel {
-        guard let url = URL(string: "\(baseUrl)/v1/login") else {
-            throw RequestError.invalidUrl
-        }
-
-        var request = URLRequest(url: url)
-        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
-        request.httpMethod = "POST"
-        request.httpBody = try? JSONEncoder().encode(LoginRequest(password: password, username: username))
-
-        guard let (data, response) = try? await URLSession.shared.data(for: request) else {
-            throw RequestError.serverError
-        }
-
-        guard let response = response as? HTTPURLResponse else {
-            throw RequestError.responseError
-        }
-
-        guard (200...299).contains(response.statusCode) else {
-            switch response.statusCode {
-            case 400:
-                throw RequestError.badRequest
-            case 401:
-                throw RequestError.unauthorized
-            case 403:
-                throw RequestError.forbidden
-            case 404:
-                throw RequestError.notFound
-            default:
-                throw RequestError.unknownError
-            }
-        }
-
-        guard let value = try? JSONDecoder().decode(LoginResponseClientModel.self, from: data) else {
-            throw RequestError.dataDecodingError
-        }
-
-        return value
+        return try await networkClient.executeRequest(
+            path: path,
+            method: .post,
+            header: ["Content-Type": "application/json"],
```

Comment:
Can the default value of header be `["Content-Type": "application/json"]`? It will be correct for 99% of HTTP calls.

Reply:
Ovaj dio sam ostavila isti jer jedino login ne koristi token. 
Ali sam napravila novu fju koja kao argument ne prima header, nego ima defaultni header koji sadrži field-ove Content-Type i Authorization.